### ISSUE TITLE
System.ComponentModel.EventBasedAsync.Tests.BackgroundWorkerTests.TestFinalization failed on mono interpreter mode

### ISSUE THREAD

**Issue by safern** [dotnet/runtime#37136]
### Console Log Summary
```
    System.ComponentModel.EventBasedAsync.Tests.BackgroundWorkerTests.TestFinalization [FAIL]
      Assert.True() Failure
      Expected: True
      Actual:   False
      Stack Trace:
        /_/src/libraries/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs(343,0): at System.ComponentModel.EventBasedAsync.Tests.BackgroundWorkerTests.TestFinalization()
        /_/src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs(359,0): at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

### Builds
|Build|Pull Request | Test Failure Count|
| --- | --- | --- |
|[#662934](https://dev.azure.com/dnceng/public/_build/results?buildId=662934)|Rolling|2|
### Configurations
- net5.0-Linux-Release-x64-mono_interpreter_release-Debian.9.Amd64.Open
### Helix Logs
|Build|Pull Request|Console|Core|Test Results|Run Client|
| --- | --- | --- | --- | --- | --- |
|[#662934](https://dev.azure.com/dnceng/public/_build/results?buildId=662934)|Rolling|[console.log](https://helix.dot.net/api/2019-06-17/jobs/a99398ee-68ee-4b99-bb65-cecbac2935c7/workitems/System.ComponentModel.EventBasedAsync.Tests/files/console.9b84e272.log)||[testResults.xml](https://helix.dot.net/api/2019-06-17/jobs/a99398ee-68ee-4b99-bb65-cecbac2935c7/workitems/System.ComponentModel.EventBasedAsync.Tests/files/testResults.xml)|[run_client.py](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-a99398ee68ee4b99bb/System.ComponentModel.EventBasedAsync.Tests/bb1e8bbc-3bfd-43cb-8d5a-5ed2e42bb7c4.log?sv=2019-02-02&se=2020-06-17T20%3A29%3A56Z&sr=c&sp=rl&sig=84xpMN9%2B3Mpha5Lk%2BRz9TSweN1UuJsnc162S5u6xTHQ%3D)|
|[#662934](https://dev.azure.com/dnceng/public/_build/results?buildId=662934)|Rolling|[console.log](https://helix.dot.net/api/2019-06-17/jobs/fcab4644-c599-46c5-b978-4418d49b6b7f/workitems/System.ComponentModel.EventBasedAsync.Tests/files/console.a87b4a5f.log)||[testResults.xml](https://helix.dot.net/api/2019-06-17/jobs/fcab4644-c599-46c5-b978-4418d49b6b7f/workitems/System.ComponentModel.EventBasedAsync.Tests/files/testResults.xml)|[run_client.py](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-fcab4644c59946c5b9/System.ComponentModel.EventBasedAsync.Tests/b09e7355-a0dd-48d7-8f8d-3bb44a62ec2a.log?sv=2019-02-02&se=2020-06-17T21%3A45%3A36Z&sr=c&sp=rl&sig=YZu3xRg0qSUU6EyPYVJ3TtaXAp9bvyf61d550fsxSy4%3D)|

cc: @steveisok

**Comment by safern** [comment]
I'll put up a PR to disable the test on interpreter mode.

**Comment by BrzVlad** [comment]
This is not really interpreter related. This is due to conservative pinning and stack pinning just happens in different patterns when using the interpreter, compared to jit code. We should disable this test based on `[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsPreciseGcSupported))]`. As other tests that call `GC.WaitForPendingFinalizers ()`

**Comment by safern** [comment]
Thanks for looking into this @BrzVlad -- makes sense.

**Comment by BrzVlad** [comment]
This is a known limitation of mono gc and we disabled the test on CI.

**Comment by safern** [comment]
@BrzVlad the test still has “ActiveIssue” attributes against this issue. We shouldn’t close issues when we add “ActiveIssue”. Instead we should skip it properly as you suggested above.

Re-opening to properly condition the test.


https://github.com/dotnet/runtime/blob/9ccfc40ec9db8bbc9806e8d4fce71b27eb442503/src/libraries/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs#L331

**Comment by BrzVlad** [comment]
oh, sorry. I thought we skipped it, will add change

**Comment by safern** [comment]
Fixed in: #38615
